library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -38,22 +39,30 @@
 
     private final ContentResolver resolver;
     private final DownloadDeleter downloadDeleter;
+    private final Uri batchContentUri;
+    private final Uri allDownloadsContentUri;
 
-    public BatchRepository(ContentResolver resolver, DownloadDeleter downloadDeleter) {
+    public static BatchRepository newInstance(ContentResolver resolver, DownloadDeleter downloadDeleter) {
```

Comment:
I guess the smell is that Downloads.Impl does work in generating its static finals and this makes anything that touches that class difficult to unit test :/